plugins/hls-alternate-number-format-plugin/src/Ide/Plugin/AlternateNumberFormat.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+
+module Ide.Plugin.AlternateNumberFormat where
+
+import           Control.Lens                         ((^.))
+import           Control.Monad.Except                 (ExceptT, MonadIO, liftIO)
+import qualified Data.HashMap.Strict                  as HashMap
+import           Data.Maybe                           (fromMaybe, isJust)
+import           Data.Text                            (Text)
+import qualified Data.Text                            as T
+import           Debug.Trace                          (traceM)
+import           Development.IDE                      (GetParsedModule (GetParsedModule),
+                                                       IdeState, RuleResult,
+                                                       Rules, define, ideLogger,
+                                                       realSrcSpanToRange,
+                                                       runAction, use,
+                                                       useWithStale)
+import           Development.IDE.Core.PositionMapping (PositionMapping)
+import           Development.IDE.GHC.Compat           hiding (getSrcSpan)
+import           Development.IDE.GHC.Compat.Util      (toList)
+import           Development.IDE.Graph.Classes        (Hashable, NFData)
+import           Development.IDE.Types.Logger         as Logger
+import           GHC.Generics                         (Generic)
+import           Ide.Plugin.Conversion                (FormatType,
+                                                       alternateFormat,
+                                                       toFormatTypes)
+import           Ide.Plugin.Literals                  (Literal (..),
+                                                       collectLiterals,
+                                                       getSrcSpan, getSrcText)
+import           Ide.Plugin.Retrie                    (handleMaybe,
+                                                       handleMaybeM, response)
+import           Ide.Types
+import           Language.LSP.Types
+import           Language.LSP.Types.Lens              (uri)
+
+descriptor :: PluginId -> PluginDescriptor IdeState
+descriptor plId = (defaultPluginDescriptor plId)
+    { pluginHandlers = mkPluginHandler STextDocumentCodeAction codeActionHandler
+    , pluginRules = collectLiteralsRule
+    }
+
+data CollectLiterals = CollectLiterals
+                     deriving (Show, Eq, Generic)
+
+instance Hashable CollectLiterals
+instance NFData CollectLiterals
+
+type instance RuleResult CollectLiterals = CollectLiteralsResult
+
+data CollectLiteralsResult = CLR {
+    literals      :: [Literal]
```

Comment:
Since we are at the nitpicking stage, I would suggest that the formatting here matches the one from: https://github.com/haskell/haskell-language-server/pull/2350/files#diff-9239cafe7a929f7c038ea0f8e32efc75af52e0f736e501259a3f7848c508e041R42

E.g.
```suggestion
data CollectLiteralsResult = CLR
    { literals      :: [Literal]
```

Reply:
Updated.